2352.ts

```diff
@@ -0,0 +1,14 @@
+function equalPairs(grid: number[][]): number {
+    let transposed: number[][] =  grid[0].map((col, c) => grid.map((row, r) => grid[r][c]))
+    let matches: number = 0
+
+    for (let i = 0; i < grid.length; i++) {
+        for (let k = 0; k < transposed.length; k++) {
+            if (grid[i].toString() === transposed[k].toString()) {
```

Comment:
This is a neat way to do this - and perfect for small scale number arrays. 

To make this more performant though, you could use a small util function such as:
```
function areArraysEqual<T>(arr1: T[], arr2: T[]): boolean {
    for (let i = 0; i < arr1.length; i ++) {
        if (arr1[i] !== arr2[i]) {
            return false;
        }
    }

    return true;
}
```

And then use that here:
```
    for (let i = 0; i < grid.length; i++) {
        for (let k = 0; k < transposed.length; k++) {
            if (areArraysEqual(grid[i], transposed[k])) {
                matches += 1;
            }
        }
    }
```